mainwindow.cc

```diff
@@ -2210,40 +2207,27 @@ void MainWindow::editPreferences()
       applyQtStyleSheet( p.addonStyle, p.displayStyle, p.darkMode );
     }
 
-    // see if we need to reapply articleview style
-    if( cfg.preferences.displayStyle != p.displayStyle ||
-      cfg.preferences.addonStyle != p.addonStyle ||
-      cfg.preferences.darkReaderMode != p.darkReaderMode )
-    {
-      articleMaker.setDisplayStyle( p.displayStyle, p.addonStyle );
-      needReload = true;
-    }
-
-    if( cfg.preferences.collapseBigArticles != p.collapseBigArticles
-        || cfg.preferences.articleSizeLimit != p.articleSizeLimit )
-    {
-      articleMaker.setCollapseParameters( p.collapseBigArticles, p.articleSizeLimit );
-    }
-
-    // See if we need to reapply expand optional parts mode
-    if( cfg.preferences.alwaysExpandOptionalParts != p.alwaysExpandOptionalParts )
-    {
-      needReload = true;
-    }
-
     // See if we need to change help language
     if( cfg.preferences.helpLanguage != p.helpLanguage )
       closeGDHelp();
 
+    // Loop through all tabs and reload pages due to ArticleMaker's change.
     for( int x = 0; x < ui.tabWidget->count(); ++x )
     {
       ArticleView & view =
         dynamic_cast< ArticleView & >( *( ui.tabWidget->widget( x ) ) );
 
       view.setSelectionBySingleClick( p.selectWordBySingleClick );
 
-      if( needReload )
-        view.reload();
+      if( cfg.preferences.displayStyle != p.displayStyle
+          || cfg.preferences.addonStyle != p.addonStyle
+          || cfg.preferences.darkReaderMode != p.darkReaderMode
+          || cfg.preferences.collapseBigArticles != p.collapseBigArticles
+          || cfg.preferences.articleSizeLimit != p.articleSizeLimit
+          || cfg.preferences.alwaysExpandOptionalParts != p.alwaysExpandOptionalParts // DSL format's special feature
```

Comment:
can extract a variable outside of the loop.

Reply:
:sweat_smile: extracted.